accounts/views.py

```diff
@@ -1,3 +1,29 @@
-from django.shortcuts import render
+from django.views.generic import TemplateView,CreateView
+from .forms import SignUpForm
+from django.urls import reverse_lazy
+from django.contrib.auth import authenticate,login
 
-# Create your views here.
+
+class SignUpView(CreateView):
+    template_name = 'accounts/signup.html'
+    form_class = SignUpForm
+    #クラスベースビューでURL解決する時に使う(renderの代わり？)
+    success_url = reverse_lazy('accounts:home')
+
+    def form_valid(self,form):
+        response = super().form_valid(form)
+        #formの情報を取得
+        username = form.cleaned_data['username']
+        password = form.cleaned_data['password1']
+        user = authenticate(self.request, username=username, password=password)
+        if user is not None:
+            login(self.request, user)
+            #リダイレクト
+            return response 

```

Comment:
```suggestion
    success_url = reverse_lazy('accounts:home')

    def form_valid(self, form):
        response = super().form_valid(form)
        username = form.cleaned_data['username']
        password = form.cleaned_data['password1']
        user = authenticate(self.request, username=username, password=password)
        if user is not None:
            login(self.request, user)
            return response 
```
コードに関係のないコメントは削除してください。

Reply:
修正しました。